dist/samples/places-autocomplete-addressform/index.js

```diff
@@ -36,42 +48,56 @@ function initAutocomplete() {
 function fillInAddress() {
   // Get the place details from the autocomplete object.
   const place = autocomplete.getPlace();
-
-  for (const component in componentForm) {
-    document.getElementById(component).value = "";
-    document.getElementById(component).disabled = false;
-  }
+  let address1 = "";
+  let postcode = "";
 
   // Get each component of the address from the place details,
   // and then fill-in the corresponding field on the form.
   for (const component of place.address_components) {
     const addressType = component.types[0];
 
-    if (componentForm[addressType]) {
-      const val = component[componentForm[addressType]];
-      document.getElementById(addressType).value = val;
+    switch (addressType) {
+      case "street_number": {
+        address1 = component[componentLength[addressType]] + " " + address1;
+        break;
+      }
+
+      case "route": {
+        address1 += component[componentLength[addressType]];
+        break;
+      }
+
+      case "postal_code": {
+        postcode = component[componentLength[addressType]] + postcode;
+        break;
+      }
+
+      case "postal_code_suffix": {
+        postcode += "-" + component[componentLength[addressType]];
+        break;
+      }
+
+      default: {
+        if (componentLength[addressType]) {
+          const val = component[componentLength[addressType]];
+          document.getElementById(addressType).value = val;
```

Comment:
you can get rid of the ids and just use 'name'

try here instead: 
```
document.querySelector(`[name=${addressType}]`).value = val;
```

Reply:
(you don't want ids because ids have to be page-unique, which could cause collisions on a customer page)